alacritty/src/event.rs

```diff
@@ -635,38 +635,46 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
             return;
         }
 
-        match &hint.action {
+        let (target, bounds) = match hint {
+            HintMatch::Hyperlink { hyperlink, .. } => (hyperlink.uri().to_owned(), None),
+            HintMatch::Regex { bounds, .. } => {
+                (self.terminal.bounds_to_string(*bounds.start(), *bounds.end()), Some(bounds))
+            },
+        };
+
+        match &hint.action() {
             // Launch an external program.
             HintAction::Command(command) => {
-                let text = self.terminal.bounds_to_string(*hint.bounds.start(), *hint.bounds.end());
                 let mut args = command.args().to_vec();
-                args.push(text);
+                args.push(target);
                 start_daemon(command.program(), &args);
             },
             // Copy the text to the clipboard.
             HintAction::Action(HintInternalAction::Copy) => {
-                let text = self.terminal.bounds_to_string(*hint.bounds.start(), *hint.bounds.end());
-                self.clipboard.store(ClipboardType::Clipboard, text);
+                self.clipboard.store(ClipboardType::Clipboard, target);
             },
             // Write the text to the PTY/search.
             HintAction::Action(HintInternalAction::Paste) => {
-                let text = self.terminal.bounds_to_string(*hint.bounds.start(), *hint.bounds.end());
-                self.paste(&text);
+                self.paste(&target);
```

Comment:
If possible we should remove the `=> { .. }` and inline stuff just to make things shorter. I think that would be possible here.